File: integration-tests/android/src/main/java/com/example/AppComponent.java
package com.example;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.DispatchingAndroidInjector;

@Component(modules = {
        AndroidInjectionModule.class,
        ActivityBindingModule.class
})
interface AppComponent {

    void inject(ExampleApp app);

    DispatchingAndroidInjector<Object> androidInjector();
}

File: integration-tests/android/src/main/java/com/example/AppComponent.kt
package com.example

import dagger.Component
import dagger.android.AndroidInjectionModule

import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidInjectionModule::class,
    ActivityBindingModule::class,
    StringModule::class]
)
internal interface AppComponent {
  //  fun androidInjector(): DispatchingAndroidInjector<Activity>

    fun inject(exampleApp: ExampleApp)
}
